oshmem/mca/spml/ucx/spml_ucx_component.c

```diff
@@ -458,7 +458,8 @@ static void mca_spml_ucx_ctx_fini(mca_spml_ucx_ctx_t *ctx)
 
 static int mca_spml_ucx_component_fini(void)
 {
-    int fenced = 0, i;
+    volatile int fenced = 0;
```

Comment:
why needed?

Reply:
`volatile` indicates that variable may be changed from another place, and therefore prevents compiler from optimizing the following loop:
```
    while (!fenced) {
...
    }
``` 
Inside the loop `fenced` variable is never modified, so in principle compiler can optimize this check, so that it's performed only once. `volatile` forces compiler to load value from the memory each time 

Btw, this is properly done in another place for the same purpose:
https://github.com/open-mpi/ompi/blob/main/opal/mca/common/ucx/common_ucx.c#L403

Reply:
right, missed that it can be updated by nb fence later